source/internal.d.ts

```diff
@@ -87,6 +79,94 @@ export type IsPlainObject<T> =
 			? true
 			: false;
 
+/**
+Converts a numeric string to a number.
+
+@example
+```
+type PositiveInt = StringToNumber<'1234'>; // => 1234;
+type NegativeInt = StringToNumber<'-1234'>; // => -1234;
+type PositiveFloat = StringToNumber<'1234.56'>; // => 1234.56;
+type NegativeFloat = StringToNumber<'-1234.56'>; // => -1234.56;
+type PositiveInfinity = StringToNumber<'Infinity'>; // => Infinity
+type NegativeInfinity = StringToNumber<'-Infinity'>; // => -Infinity
+```
+
+@category String
+@category Template literal
+*/
+export type StringToNumber<S extends string> = S extends `${infer N extends number}`
+	? N
+	: S extends 'Infinity'
+		? PositiveInfinity
+		: S extends '-Infinity'
+			? NegativeInfinity
+			: never;
+
+/**
+Returns a boolean for whether the string `S` starts with `SearchString`
+
+@example
+```
+StartsWith<'abcde', 'abc'>
+//=> true
+
+StartsWith<'abcde', 'bc'>
+//=> false
+
+StartsWith<string, 'bc'>
+//=> boolean
+
+StartsWith<'abcde', string>
+//=> boolean
+```
+
+@category String
+@category Template literal
+*/
+export type StartsWith<S extends string, SearchString extends string> = string extends S | SearchString
+	? boolean
```

Comment:
How about reurn `never`?

Reply:
return `boolean` is better i think :), eg:
```tsx
type T = StartsWith<string, 'a'>;
```
a `string` can be `abc`, then `T` should be `true`;
and `string` can be `bc`, then `T` should be false;

see also: https://github.com/microsoft/TypeScript/issues/51090#issuecomment-1272357947